### ISSUE TITLE
Stacktrace in resque web

### ISSUE THREAD

**Issue by wr0ngway** [ryanlecompte/redis_failover#14]
Another resque web related stack trace:

```
NoMethodError - undefined method `[]' for nil:NilClass:
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:328:in `verify_role!'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:230:in `master'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:208:in `dispatch'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:82:in `block (2 levels) in <class:Client>'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-namespace-1.0.3/lib/redis/namespace.rb:213:in `method_missing'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/resque-dynamic-queues-0.7.1/lib/resque/plugins/dynamic_queues/attributes.rb:36:in `block in set_dynamic_queues'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-2.2.2/lib/redis.rb:1015:in `block (2 levels) in multi'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-2.2.2/lib/redis.rb:978:in `block in pipelined'
    /usr/local/rvm/rubies/ruby-1.9.3-p125-falcon/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-2.2.2/lib/redis.rb:975:in `pipelined'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-2.2.2/lib/redis.rb:1013:in `block in multi'
    /usr/local/rvm/rubies/ruby-1.9.3-p125-falcon/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-2.2.2/lib/redis.rb:1009:in `multi'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:208:in `dispatch'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis_failover-0.8.2/lib/redis_failover/client.rb:82:in `block (2 levels) in <class:Client>'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/redis-namespace-1.0.3/lib/redis/namespace.rb:184:in `method_missing'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/resque-dynamic-queues-0.7.1/lib/resque/plugins/dynamic_queues/attributes.rb:35:in `set_dynamic_queues'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/resque-dynamic-queues-0.7.1/lib/resque/plugins/dynamic_queues/server.rb:46:in `block in registered'
    /mnt/appleton-matt/shared/bundle/ruby/1.9.1/gems/sinatra-1.3.2/lib/sinatra/base.rb:1212:in `call'
```

**Comment by ryanlecompte** [comment]
That's interesting. This looks like somehow the underlying redis client is returning nil as a response to an #info command. I've never seen that before. I wonder if this has something to do with the resque-dynamic-queues and it perhaps monkey patching something. Does it happen consistently?

**Comment by wr0ngway** [comment]
Its repeatable and happens when using the redis multi command:

```
[1] pry(main)> r = RedisFailover::Client.new(:zkservers => "localhost:2181")    => #<RedisFailover::Client (master: localhost:6379, slaves: none)>
[2] pry(main)> r.multi do
[2] pry(main)*   r.del("foo")
[2] pry(main)*   r.del("bar")
[2] pry(main)* end  
NoMethodError: undefined method `[]' for nil:NilClass
from /Users/mconway/.rvm/gems/ruby-1.9.3-p125-falcon/bundler/gems/redis_failover-641b30707a5a/lib/redis_failover/client.rb:320:in `verify_role!'
```

**Comment by ryanlecompte** [comment]
Fixed in a20d3ca.
